src/geode/geometry/intersection_detection.cpp

```diff
@@ -91,23 +99,65 @@ namespace
                         triangle_points_projection[edge_v0 == 2
                                                        ? 0
                                                        : edge_v0 + 1] } );
-            if( seg_edge_inter.first != geode::Position::outside
-                && seg_edge_inter.first != geode::Position::parallel )
+            if( seg_edge_inter.first == geode::Position::outside
+                || seg_edge_inter.first == geode::Position::parallel )
+            {
+                continue;
+            }
+            if( ( seg_edge_inter.first == geode::Position::vertex0
+                    || seg_edge_inter.first == geode::Position::vertex1 )
+                && ( seg_edge_inter.second == geode::Position::vertex0
+                     || seg_edge_inter.second == geode::Position::vertex1 ) )
             {
-                return { geode::Position::parallel, geode::Position::parallel };
+                result.first = seg_edge_inter.first;
+                if( seg_edge_inter.second == geode::Position::vertex0 )
+                {
+                    if( edge_v0 == 0 )
```

Comment:
You could create a static const array of Position to avoid the if/elseif/else block